File: Practica-POO/Semana01/Ejercicio10/Game.cpp
#include <iostream>
#include <string>
using namespace std;

class Game{
    public:
    string name;
    double healtpoints;
    double strengs;
    bool state;

    Game(string n , double h , double s){
        name = n;
        healtpoints = h;
        strengs = s;
        state = true;
    }

    string RecibirDanio(double d){
        healtpoints -= d;
        return "Has recibido daño -->" + to_string(d) + " puntos de vida";
    }

    string Healt(double h){
        healtpoints += h;
        return "Has recibido curacion --> " + to_string(h) + " puntos de vida";
    }
    string State(){
        if(healtpoints <=0){
            state = false;
            return "Moriste";
        }else{
            state = true;
            return "Sigues vivo";
        }
    }
};

int main(){
    Game game("Jeremy", 400,70);

    double ataque1= game.RecibirDanio(50);

    double healt1 = gamte.Healt(60);

    cout << ataque1 << endl;
    cout << healt1 << endl;

    return 0; 
}

File: Practica-POO/Semana01/Ejercicio10/Game.py
class Game:
    def __init__(self,name,healtpoints,streng):
        self.name = name 
        self.healtpoints = healtpoints
        self.streng = streng
        self.estadu = True

    def RecibirAtaque(self,daño):
        self.healtpoints -= daño
        return f"Recibiste --> {daño} puntos de daño"

    def Healt(self,healt):
        self.healtpoints +=healt
        return f"Recibiste curacion --> {healt} puntos de vida"

    def estado(self):
        if(self.healtpoints>0):
            self.estadu = True
            return "Sigues vivo"
        else:
            self.estadu = False
            return "Moriste"

person1 = Game("Jeremy",100,20)
ataque1=person1.RecibirAtaque(40)
healt1 = person1.Healt(50)


